### PR TITLE
expose generateSpecAndRoutes for consumption

### LINKED ISSUE #964: Can generateSpecAndRoutes please be a module export?
Hello, I'm wondering if you would mind exposing/re-exporting the generateSpecAndRoutes function? And also allow me to pass in the configuration object manually at this point (ie, ConfigArgs referenced by generateSpecAndroutes would be of type `string | Config`? I'd like to bundle this in with my existing build system and it would be convenient for this to be directly exposed.

I'm already watching my filesystem, and the rest of my build process and build configuration (using esbuild) is colocated. It would help consolidate both process and documentation for me to be able to codify in the same place as everything else in our build that:

"Whenever these controller files change, run generateSpecAndRoutes". I could just replicate what you're already doing with that function, but it would be nice to gain ongoing compatibility with whatever updates/improvements/changes you end up making down the road.

## Sorting

- **I'm submitting a ...**

  - [ ] bug report
  - [X] feature request
  - [ ] support request

- I confirm that I
  - [X] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

```ts
import { generateSpecAndRoutes } from 'tsoa';

generateSpecAndRoutes({
  configuration: { 
    routes: {
      routesDir: "src/server"
      //...etc
     },
    spec: {
      specVersion: 3 
      // ...etc
    }
  }
});
```

## Current Behavior

```ts
import { generateSpecAndRoutes } from 'tsoa' // no exported member
```

## Steps to Reproduce

1. 2. 3. 4.

## Context (Environment)

Version of the library: 
Version of NodeJS: 

- Confirm you were using yarn not npm: [ ]

### PR DESCRIPTION
**Pull request by mscottnelson**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [ ] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

**Closing issues**

closes #964

### If this is a new feature submission:

- [x] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

**Test plan**

### CODE REVIEW

File: packages/cli/src/cli.ts [hunk 1]
```diff
@@ -74,6 +74,10 @@ const getConfig = async (configPath = 'tsoa.json'): Promise<Config> => {
   return config;
 };
 
+const resolveConfig = async (config?: string | Config) => {
+  return typeof config === 'object' ? config : await getConfig(config);
```
Comment (WoH) [inline packages/cli/src/cli.ts:78]: Don't release Zalgo

Comment (mscottnelson) [reply packages/cli/src/cli.ts]: I know about Zalgo but I'm not sure what you're asking for here. This is an async function, and its return value will always be on "a later tick". 

Returning `config` directly will still result in an async function signature / Promise. Adding an await, as in `return await config` *will* further defer by adding an additional Promise resolution, but it doesn't increase determinacy in any way that I can see? For that matter, I guess `getConfig` doesn't need to be awaited, either, and `return (typeof config==="config") ? config : getConfig(config)` would remove an unneeded intermediary Promise resolution.

I added a clarifying return type to the function signature. Let me know if there's something else I am missing here. Thanks for looking this over!

Comment (WoH) [reply packages/cli/src/cli.ts]: Missed the async for some reason, your right

**Comment by WoH** [comment]
> If you think it aligns, I'd be happy to add unit tests to cover the various scenarios.

Can you add 1/2 of these?

**Comment by WoH** [comment]
I'll merge this as is, given it affects the API only in places where we explicitly state that we don't follow SemVer there.

**Comment by mscottnelson** [comment]
thanks @WoH, apologies that I didn't get an opportunity to come back around to the tests.

**Pull request merged**